code/search/binary_search/binary_search.c

```diff
@@ -2,46 +2,40 @@
 /*
  * Part of Cosmos by OpenGenus Foundation
 */
-int recursiveBinarySearch(int arr[], int l, int r, int x)
-{
-   if (r >= l)
-   {
-        int mid = l + (r - l)/2;
- 
-        if (arr[mid] == x)  
-            return mid;
-         if (arr[mid] > x) 
-             return recursiveBinarySearch(arr, l, mid-1, x);
-       
-         return recursiveBinarySearch(arr, mid+1, r, x);
-   }
-   return -1;
-}
-
 int binarySearch(int arr[], int l, int r, int x)
```

Comment:
It already seems to have an iterative version implemented, and your version is equivalent to it.

Reply:
You can make an update to add comments to explain the code, but if so you should just add the new information.

Reply:
sure